pvlib/iotools/pvgis.py

```diff
@@ -23,6 +23,346 @@
+
+    PVGIS data is freely available at [1]_.
+
+    Parameters
+    ----------
+    latitude: float
+        In decimal degrees, between -90 and 90, north is positive (ISO 19115)
+    longitude: float
+        In decimal degrees, between -180 and 180, east is positive (ISO 19115)
+    start: int or datetime like, default: None
+        First year of the radiation time series. Defaults to first year
+        available.
+    end: int or datetime like, default: None
+        Last year of the radiation time series. Defaults to last year
+        available.
+    raddatabase: str, default: None
+        Name of radiation database. Options depend on location, see [3]_.
+    components: bool, default: True
+        Output solar radiation components (beam, diffuse, and reflected).
+        Otherwise only global irradiance is returned.
+    surface_tilt: float, default: 0
+        Tilt angle from horizontal plane. Not relevant for 2-axis tracking.
+    surface_azimuth: float, default: 0
+        Orientation (azimuth angle) of the (fixed) plane. 0=south, 90=west,
+        -90: east. Not relevant for tracking systems.
+    usehorizon: bool, default: True
+        Include effects of horizon
+    userhorizon: list of float, default: None
+        Optional user specified elevation of horizon in degrees, at equally
+        spaced azimuth clockwise from north, only valid if `usehorizon` is
+        true, if `usehorizon` is true but `userhorizon` is `None` then PVGIS
+        will calculate the horizon [4]_
+    pvcalculation: bool, default: False
+        Return estimate of hourly PV production.
+    peakpower: float, default: None
+        Nominal power of PV system in kW. Required if pvcalculation=True.
+    pvtechchoice: {'crystSi', 'CIS', 'CdTe', 'Unknown'}, default: 'crystSi'
+        PV technology.
+    mountingplace: {'free', 'building'}, default: free
+        Type of mounting for PV system. Options of 'free' for free-standing
+        and 'building' for building-integrated.
+    loss: float, default: 0
+        Sum of PV system losses in percent. Required if pvcalculation=True
+    trackingtype: {0, 1, 2, 3, 4, 5}, default: 0
+        Type of suntracking. 0=fixed, 1=single horizontal axis aligned
+        north-south, 2=two-axis tracking, 3=vertical axis tracking, 4=single
+        horizontal axis aligned east-west, 5=single inclined axis aligned
+        north-south.
+    optimal_surface_tilt: bool, default: False
+        Calculate the optimum tilt angle. Not relevant for 2-axis tracking
+    optimalangles: bool, default: False
+        Calculate the optimum tilt and azimuth angles. Not relevant for 2-axis
+        tracking.
+    outputformat: str, default: 'json'
+        Must be in ``['json', 'csv']``. See PVGIS hourly data
+        documentation [2]_ for more info.
+    url: str, default: const:`pvlib.iotools.pvgis.URL`
+        Base url of PVGIS API. ``seriescalc`` is appended to get hourly data
+        endpoint.
```

Comment:
@kanderso-nrel Would it be preferred here to also have the URL as a module variable instead of an input?

Reply:
Oh, maybe.  Pro: it would shorten a long parameter list.  Con: inconsistent with the PVGIS TMY function.  If you do remove it and make it a module variable, it should be named such that there is room to eventually make the PVGIS TMY URL a module variable as well.

Reply:
PVGIS hourly and TMY share the same base URL, so the change would have to be made to both at the same time. I'm ok for leaving it for now though